### PR TITLE
Fix company contacts list

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for 4.0
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

#### Description:
If company has not assigned contacts, it show wrong contacts.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create contacts
3. Create companies
4. Check detail of company without contact
5. In some cases you see contacts, even If we do not expect it
6. This PR fixed it and added note about no contacts assigned to company

image

### CODE REVIEW

**Comment by dadarya0** [comment]
@kuzmany Your fix is good no issue in code but I am not able to reproduce above mentioned issue in feature branch.

Screenshot 2021-06-07 at 4 23 24 PM

I am getting `No Results Found` message if no contact is associated with company.

CC - @RCheesley

File: app/bundles/LeadBundle/Translations/en_US/messages.ini [hunk 1]
```diff
@@ -780,6 +780,7 @@ mautic.channel.stat.leadcount.tooltip="Details may not match summary numbers if
 mautic.lead.segments.usages="Segment usages"
 mautic.lead.campaign.share="Campaign share"
 mautic.lead.share="Share"
+mautic.company.contact.is.not.associated="No contact assigned to this company."
```
Comment (RCheesley) [inline app/bundles/LeadBundle/Translations/en_US/messages.ini]:
```suggestion
mautic.company.contact.is.not.associated="No contacts are assigned to this company."
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
I too am unable to reproduce this bug with the features branch - perhaps you could give us some examples of when this issue is being triggered @kuzmany ?

Small grammar suggestion for the language string.

**Comment by HN-PRO** [comment]
> I too am unable to reproduce this bug with the features branch - perhaps you could give us some examples of when this issue is being triggered @kuzmany ?
> 
> Small grammar suggestion for the language string.

I reproduced the problemin version 3.3.3. This problem will occur if the company does not have any contacts and there are ids characters in the firstname, lastname, email, company, city, state, zipcode, and country fields in the contact.
The following is the SQL I debugged, FYI.
`SELECT l.* FROM leads l LEFT JOIN users u ON u.id = l.owner_id 
WHERE (l.firstname LIKE :MnNcgRPr) OR (l.lastname LIKE :MnNcgRPr) OR (l.email LIKE :MnNcgRPr) OR (l.company LIKE :MnNcgRPr) 
OR (l.city LIKE :MnNcgRPr) OR (l.state LIKE :MnNcgRPr) OR (l.zipcode LIKE :MnNcgRPr) OR (l.country LIKE :MnNcgRPr) 
ORDER BY l.last_active DESC LIMIT 30`

**Comment by RCheesley** [comment]
@HN-PRO did this patch fix the issue for you?  If so we have a code review and test and can look at merging it.

**Comment by HN-PRO** [comment]
> @HN-PRO did this patch fix the issue for you? If so we have a code review and test and can look at merging it.

Yes, it has helped me solve the problem. @RCheesley

**Comment by kuzmany** [comment]
@RCheesley seems like two good reviews here

**Comment by RCheesley** [comment]
@kuzmany indeed but we need test coverage - it is [currently failing](https://github.com/mautic/mautic/pull/10125/checks?check_run_id=3181404085) on that.

**Comment by RCheesley** [comment]
Bumping from release due to inadequate automated test code coverage.

**Comment by RCheesley** [comment]
Last milestone bump before closing the PR - @kuzmany if you are able to rebase to 4.x and look at the test before the 4.2 release we can consider it for the release.

**Comment by mollux** [comment]
Closing in favor of #12464

**Pull request closed without merging**